crates/common/src/fs.rs

```diff
@@ -97,6 +97,20 @@ pub fn write(path: impl AsRef<Path>, contents: impl AsRef<[u8]>) -> Result<()> {
     fs::write(path, contents).map_err(|err| FsPathError::write(err, path))
 }
 
+/// Write and attempt to ensure that all in-memory data reaches the filesystem before returning.
+pub fn write_sync(path: impl AsRef<Path>, contents: impl AsRef<[u8]>) -> Result<()> {
+    let path = path.as_ref();
+    let mut file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(path)
+        .map_err(|err| FsPathError::write(err, path))?;
+    file.write_all(contents.as_ref())
+        .and_then(|_| file.sync_all())
+        .map_err(|err| FsPathError::write(err, path))
```

Comment:
nit: the dual use of `FsPathError::write` can be confusing here as the first one refers to an error related to the opening of the file and not the writing